cmd/snap-confine/snap-confine.apparmor.in

```diff
@@ -535,6 +535,31 @@
         /var/snap/{,*/} r,
         /var/snap/*/**/ rw,
 
+        # Allow creating placeholder directory in /tmp/.snap/ as support for
+        # the writable mimic code that can poke holes in arbitrary read-only
+        # places using tmpfs and bind mounts.
+        /tmp/ r,
+        /tmp/.snap/ rw,
+        /tmp/.snap/** rw,
+        # Allow mounting/unmounting any part of $SNAP over to a temporary place
+        # in /tmp/.snap/ during the preparation of a writable mimic.
+        mount options=(bind, rw) /snap/*/** -> /tmp/.snap/**,
+        # Allow mounting tmpfs over the original $SNAP/** directory.
+        mount fstype=tmpfs options=(rw) tmpfs -> /snap/*/**,
+        # Allow bind mounting anything from the temporary place in /tmp/.snap/
+        # back to $SNAP/** (to re-construct the data that was there before).
+        mount options=(bind, rw) /tmp/.snap/** -> /snap/*/**,
+        # Allow unmounting the temporary directory in /tmp once it is no longer
+        # necessary.
+        umount /tmp/.snap/**,
+
+        # Allow creating missing directories under $SNAP once the writable
+        # mimic is ready.
+        / r,
+        /snap/ r,
+        /snap/*/ r,
+        /snap/*/** rw,
```

Comment:
Not a blocker. I would probably rewrite this as:
```
/snap/{,**} r,
/snap/*/** w,
```

Reply:
Applied